test/fork/PSMVariant3Actions.t.sol

```diff
@@ -0,0 +1,106 @@
+
+        assertEq(dai.balanceOf(PSM_LITE),                               PSM_DAI_START);
+        assertEq(usdc.balanceOf(pocket),                                PSM_USDC_START);
+        assertEq(usdc.balanceOf(address(this)),                         100e6);
+        assertEq(susds.convertToAssets(susds.balanceOf(address(this))), 0);
+
+        actions.swapAndDeposit(address(this), 100e6, 100e18);
+
+        assertEq(dai.balanceOf(PSM_LITE),                               PSM_DAI_START - 100e18);
+        assertEq(usdc.balanceOf(pocket),                                PSM_USDC_START + 100e6);
+        assertEq(usdc.balanceOf(address(this)),                         0);
+        assertEq(susds.convertToAssets(susds.balanceOf(address(this))), 99.999999999999999999e18);  // Rounding
+    }
+
+    function test_withdrawAndSwap() public {
+        uint256 shares = susds.convertToShares(100e18);
+        deal(SUSDS, address(this), shares);
+        susds.approve(address(actions), shares);
+
+        assertEq(dai.balanceOf(PSM_LITE),                               PSM_DAI_START);
+        assertEq(usdc.balanceOf(pocket),                                PSM_USDC_START);
+        assertEq(usdc.balanceOf(address(this)),                         0);
+        assertEq(susds.convertToAssets(susds.balanceOf(address(this))), 99.999999999999999999e18);  // Rounding
+
+        // Make slightly lower than 100e6 to account for rounding errors
+        actions.withdrawAndSwap(address(this), 99e6, 100e18);
+
+        assertEq(dai.balanceOf(PSM_LITE),                               PSM_DAI_START + 99e18);
+        assertEq(usdc.balanceOf(pocket),                                PSM_USDC_START - 99e6);
+        assertEq(usdc.balanceOf(address(this)),                         99e6);
+        assertEq(susds.convertToAssets(susds.balanceOf(address(this))), 0.999999999999999998e18);  // Some dust left over
+    }
+
+    function test_redeemAndSwap() public {
+        uint256 shares = susds.convertToShares(100e18);
+        deal(SUSDS, address(this), shares);
+        susds.approve(address(actions), shares);
+
+        assertEq(dai.balanceOf(PSM_LITE),                               PSM_DAI_START);
+        assertEq(usdc.balanceOf(pocket),                                PSM_USDC_START);
+        assertEq(usdc.balanceOf(address(this)),                         0);
+        assertEq(susds.convertToAssets(susds.balanceOf(address(this))), 99.999999999999999999e18);  // Rounding
+
+        // Make slightly lower than 100e6 to account for rounding errors
+        actions.redeemAndSwap(address(this), shares, 99e6);
+
+        assertEq(dai.balanceOf(PSM_LITE),                               PSM_DAI_START + 99.999999e18);
+        assertEq(usdc.balanceOf(pocket),                                PSM_USDC_START - 99.999999e6);
+        assertEq(usdc.balanceOf(address(this)),                         99.999999e6);
+        assertEq(susds.convertToAssets(susds.balanceOf(address(this))), 0);
```

Comment:
Do we have also 0 shares at the end, or are we left with 1 that is wort less than 1, rounded to 0?